### PR TITLE
Fix issue on older PHP versions

### LINKED ISSUE #77: PHP - Specify minimum supported PHP version?
Composer.json doesn't specify a minimum support php version.

- Check implication of not specifying a minimum version
- What's our desired minimum supported version?

part of https://github.com/std-uritemplate/std-uritemplate/issues/54

### PR DESCRIPTION
**Pull request by andreaTP**
Fix #77 and #76 

Please @Ndiritu let me know if this looks like it 🙏

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -85,7 +85,7 @@ jobs:
         uses: shivammathur/setup-php@v2
         if: matrix.language == 'php'
         with:
-          php-version: '8.2'
+          php-version: '7.4'
```
Comment (Ndiritu) [inline .github/workflows/test.yml:88]: We can use a PHP version matrix for this to cover all versions [sample](https://github.com/microsoft/kiota-abstractions-php/blob/dev/.github/workflows/pr-validation.yml#L15)

File: php/composer.json [hunk 2]
```diff
@@ -3,6 +3,9 @@
     "description": "Std UriTemplate, RFC 6570 implementation",
     "type": "library",
     "license": "Apache-2.0",
+    "require": {
+        "php": ">=7.4"
```
Comment (Ndiritu) [inline php/composer.json:7]:
```suggestion
        "php": "^7.4 || ^8.0"
```
To protect us from breaking changes post PHP 8.x

File: php/src/StdUriTemplate.php [hunk 3]
```diff
@@ -291,17 +291,17 @@ private static function isList($value) {
...
-      // https://stackoverflow.com/a/173479/7898052
-      if (!function_exists('array_is_list')) {
-        function array_is_list(array $arr)
-        {
-            if ($arr === []) {
-                return true;
+        if (is_array($value)) {
+            // https://stackoverflow.com/a/173479/7898052
+            if (!function_exists('array_is_list')) {
+                if ($value === []) {
```
Comment (Ndiritu) [inline php/src/StdUriTemplate.php:297]: 👍🏾

File: .github/workflows/test.yml (hunk 1 above)
Comment (andreaTP) [reply .github/workflows/test.yml:88]: I would like not to grow the matrix for only one language, but, in this case it's going to test 2 different code paths, so I'll add a matrix for `7.4` as min and `8.2` as latest

**Pull request merged**